src/test/scala/mesosphere/marathon/api/akkahttp/v2/GroupsControllerTest.scala

```diff
@@ -1,26 +1,97 @@
 package mesosphere.marathon
 package api.akkahttp.v2
 
+import akka.http.scaladsl.model.Uri.{ Path, Query }
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.Accept
 import akka.http.scaladsl.testkit.ScalatestRouteTest
 import com.typesafe.scalalogging.StrictLogging
 import mesosphere.UnitTest
 import mesosphere.marathon.api.TestAuthFixture
-import mesosphere.marathon.core.appinfo.GroupInfoService
+import mesosphere.marathon.api.akkahttp.Rejections.EntityNotFound
+import mesosphere.marathon.core.appinfo.{ AppInfo, GroupInfo, GroupInfoService }
 import mesosphere.marathon.core.election.ElectionService
+import mesosphere.marathon.state.PathId
+import mesosphere.marathon.state.PathId._
 import mesosphere.marathon.test.GroupCreation
 import org.scalatest.Inside
 
+import scala.concurrent.Future
+
 class GroupsControllerTest extends UnitTest with ScalatestRouteTest with Inside with RouteBehaviours with StrictLogging with GroupCreation {
-  case class Fixture(authenticated: Boolean = true,
-                     authorized: Boolean = true,
-                     authFn: Any => Boolean = _ => true,
-                     electionService: ElectionService = mock[ElectionService],
-                     infoService: GroupInfoService = mock[GroupInfoService]) {
+
+  "Group detail" should {
+    {
+      val controller = Fixture(authenticated = false).groupsController
+      behave like unauthenticatedRoute(forRoute = controller.route, withRequest = Get(Uri./).addHeader(Accept(MediaTypes.`text/plain`)))
+    }
+
+    "returns group detail" in {
+      val infoService = mock[GroupInfoService]
+      infoService.selectGroup(eq(PathId("/test/group")), any, any, any) returns Future.successful(Some(GroupInfo(createGroup(PathId("/test/group")), None, None, None)))
+      val f = new Fixture(infoService = infoService)
+
+      Get(Uri./.withPath(Path("/test/group"))) ~> f.groupsController.route ~> check {
+        status should be(StatusCodes.OK)
+        responseAs[String] should include (""""id" : "/test/group"""")
+      }
+    }
+
+    "returns empty group if root and no group found" in {
+      val infoService = mock[GroupInfoService]
+      infoService.selectGroup(any, any, any, any) returns Future.successful(None)
```

Comment:
Should we match a specific parameter here?

Reply:
I matched at least the pathid, the rest would be too much implementation details in my opinion...